src/main/java/com/basho/riak/client/core/operations/ts/FetchOperation.java

```diff
@@ -0,0 +1,123 @@
+package com.basho.riak.client.core.operations.ts;
+
+import com.basho.riak.client.core.converters.TimeSeriesPBConverter;
+import com.basho.riak.client.core.operations.PBFutureOperation;
+import com.basho.riak.client.core.query.timeseries.Cell;
+import com.basho.riak.client.core.query.timeseries.QueryResult;
+import com.basho.riak.client.core.util.BinaryValue;
+import com.basho.riak.protobuf.RiakKvPB;
+import com.basho.riak.protobuf.RiakMessageCodes;
+import com.google.protobuf.ByteString;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.List;
+
+/**
+ * An operation to fetch a single row in a Riak Time Series table.
+ *
+ * @author Alex Moore <amoore at basho dot com>
+ * @since 2.0.3
+ */
+public class FetchOperation extends PBFutureOperation<QueryResult, RiakKvPB.TsGetResp, BinaryValue>
+{
+    private static final Logger logger = LoggerFactory.getLogger(FetchOperation.class);
+    private final Builder builder;
+
+    private FetchOperation(Builder builder)
+    {
+        super(RiakMessageCodes.MSG_TsGetReq,
+              RiakMessageCodes.MSG_TsGetResp,
+              builder.reqBuilder,
+              RiakKvPB.TsGetResp.PARSER);
+
+        this.builder = builder;
+    }
+
+    @Override
+    protected QueryResult convert(List<RiakKvPB.TsGetResp> responses)
+    {
+        // This is not a streaming op, there will only be one response
+        if (responses.size() > 1)
+        {
+            logger.error("Received {} responses when only one was expected.", responses.size());
```

Comment:
I moved some common things to the base, but want to hold off on generalizing it more until later (i.e. creating a SingleResponsePBFutureOperation and MultiResponsePBFutureOperation), because that would necessitate more refactorings at this time.